modules/common/util/template_util.go

```diff
@@ -244,5 +287,14 @@ func GetTemplateData(t Template) (map[string]string, error) {
 		data[filename] = renderedTemplate
 	}
 
+	if t.PostProcessCleanup {
+		for filename, content := range data {
+			if strings.HasSuffix(filename, ".conf") {
```

Comment:
As the cleaning is now optional I think the caller can make sure it is only enabled for the files necessary so we don't need to filter for extension (which also not super specific as the current cleaning is for ini like files, but a .conf file might not be ini like)

Reply:
as a single Template represents a set of diverse files instead of a single file. My suggestion that the client can use the PostProcessCleanup boolean instead of the static .conf suffix filtering does not work. So we need to keep the filter to avoid removing lines from other than ini lines.